bark/models/behavior/rule_based/intersection_behavior.cpp

```diff
@@ -76,9 +76,9 @@ std::pair<AgentId, bool> BehaviorIntersectionRuleBased::GetIntersectingAgent(
           observed_world.GetLaneCorridor()->GetCenterLine(), ego_pos));
       double s_other = std::get<1>(GetNearestPointAndS(
           observed_world.GetLaneCorridor()->GetCenterLine(), agent_pos));
-      double ego_angle = Norm0To2PI(ego_state[THETA_POSITION]);
-      double other_angle = Norm0To2PI(agent_state[THETA_POSITION]);
-      if (fabs(ego_angle - other_angle) > angle_diff_for_intersection_ &&
+      double ego_angle = ego_state[THETA_POSITION];
+      double other_angle = agent_state[THETA_POSITION];
+      if (AngleDiff(ego_angle, other_angle) > angle_diff_for_intersection_ &&
```

Comment:
I think a signed angle diff is required here actually.
if both angles are in the range of [-pi, pi], the difference could be -0.2 or 0.2 which both would equal to 0.2.
So we loose the information of the direction which is required as we only want to yield vehicles from the right

Reply:
you are right